app/livechat/server/methods/setDepartmentForVisitor.js

```diff
@@ -22,7 +23,8 @@ Meteor.methods({
 
 		const transferData = {
```

Comment:
Since you already pass all the information you need through the Livechat. transfer, why not processing the transferData inside the Livechat. transfer?
IMO it would reduce the complexity of the algorithm.